src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -162,17 +164,108 @@ MdnsContexts MdnsContexts::sInstance;
 
 namespace {
 
+static void OnRegisterRecord(DNSServiceRef sdRef, DNSRecordRef recordRef, DNSServiceFlags flags, DNSServiceErrorType err,
+                             void * context)
+{
+    if (err != kDNSServiceErr_NoError)
+    {
+        ChipLogError(Discovery, "Mdns: %s Registered Record failed. Error: %d", __func__, err);
+        auto sdCtx = reinterpret_cast<RegisterRecordContext *>(context);
+        sdCtx->Finalize(err);
+    }
```

Comment:
Do we need the check for `if (err != kDNSServiceErr_NoError)` ?
It looks like we are going to log the error twice (with the log in `DispatchFailure`) and I think that we want `sdCtx->Finalise(err)` to be called all the time, so `DispatchSuccess` can be called and the context released properly.

Reply:
I am running into issues if i call the finalize for success case but i agree with you. if i don't do that it will result in not releasing the context. let me figure out what the problem is when i call it for success case

Reply:
fixed